src/Bicep.Core/Semantics/InternalVariableSymbol.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.Semantics
+{
+    public class InternalVariableSymbol : Symbol
```

Comment:
>Symbol [](http://example.com/codeflow?start=42&length=6)

This shouldn't be a symbol because it's not generated during binding and is not really part of the symbol hierarchy.

Reply:
Agreed. I feel like the semantic model should not care about whether a special variable needs to be created or not, and this can be an optmization step during code generation (e.g., create a new visitor to visit all loadTextContent function calls and collect a list of variables we need to emit).

Reply:
changed to just `FunctionVariable`  POCO